index.js

```diff
@@ -128,19 +128,30 @@ class PostHog {
      * @return {PostHog}
      */
 
-    capture(message, callback) {
+    async capture(message, callback) {
```

Comment:
this can have unintended consequences, as it's making an interface change where a function taking a callback is now also using async/await and thus returns a Promise.

I'm not yet sure how clients would get affected, need to test this a bit. Specially around anywhere else in this lib calling `this.capture()`

Reply:
This is a major version breaking change in the node ecosystem :).

When something is async and you don't expect it to be, it can leave all sorts of code running in the background, which might cause problems. We've been seeing a lot of issues with voided promises in the plugin server. For simpler nodejs apps, like command line scripts, they'll probably just hang for a while after exiting, etc.

Reply:
Happy to take suggestions on how to grab the feature flags as part of the capture call 😀

Reply:
They need to be fetched and cached before the call. 🤷

Reply:
Probably the not-so-nice but non-breaking way would be to just use it as a promise instead and put everything else in the `.then`  callback

Reply:
Another option might be to do what Ben says, but on the `enqueue` API (i.e. move this logic to enqueue), which is private, and open to a lot more modification, versus, `capture`, which is public.

This gets rid of the not-so-nice ness? But a bit annoying that this logic happens in enqueue

Reply:
thanks folks!!

Reply:
I ended up leaving it in capture instead of enqueue but can easily move the function call to enqueue if it makes a big difference @neilkakkar 🤗

Reply:
Even better!